### PR TITLE
Gaming update minimal

### PR DESCRIPTION
**Pull request by ZackaryHoyt**
Did not break any APIs that the students will use with this minimal update variant. Everyone should be able to pull this into their project and not break anything. Note this wouldn't update the config files, so there shouldn't be conflicts there either.

### CODE REVIEW

File: src/spacesettlers/game/GameFactory.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  */
 
 public class GameFactory {
-	public static AbstractGame generateNewGame() {
-		return new TicTacToe3D();
+	public static TicTacToe3D generateNewGame(final AbstractGameAgent lhs_player, final AbstractGameAgent rhs_player) {
+		return new TicTacToe3D(lhs_player, rhs_player);
 	}
```
Comment (amymcgovern) [inline src/spacesettlers/game/GameFactory.java:14]: Are you removing the abstract part just to make less casting?  My intent is to implement extra games eventually

File: src/spacesettlers/simulator/CollisionHandler.java [hunk 2]
```diff
@@ -495,30 +494,23 @@ public void damageAiCore(AiCore core) {
...
 		
 		if (opponent == null) {
 			System.out.println("Gaming asteroid reached and no player specified by opponent: Winner is asteroid.");
 			return false;
 		} else {
 			System.out.println("Gaming asteroid reached and proceeding with game");
 		}
+
+		HeuristicTicTacToe3DGameAgent myPlayer = new HeuristicTicTacToe3DGameAgent(0); // arg does nothing, but didn't want to break the api so it needs an int passed in
+		final TicTacToe3D game = GameFactory.generateNewGame(myPlayer, opponent);
```
Comment (amymcgovern) [inline src/spacesettlers/simulator/CollisionHandler.java]: Same comment here - it isn't abstract anymore?

**Review by amymcgovern — COMMENTED** [review]
I think it is close - just wondering about the two things I commented on

File: src/spacesettlers/game/GameFactory.java (hunk 1 above)
Comment (Zwithak) [reply src/spacesettlers/game/GameFactory.java:14]: Reverted returned type to be an abstract object.

File: src/spacesettlers/simulator/CollisionHandler.java (hunk 2 above)
Comment (Zwithak) [reply src/spacesettlers/simulator/CollisionHandler.java]: Reverted type to be an abstract object.

**Review by amymcgovern — APPROVED** [review]

**Pull request merged**